math/src/field/extensions/cubic.rs

```diff
@@ -112,6 +117,39 @@ impl<B: ExtensibleField<3>> FieldElement for CubeExtension<B> {
         Self(result[0], result[1], result[2])
     }
 
+    // BASE ELEMENT CONVERSIONS
+    // --------------------------------------------------------------------------------------------
+
+    fn base_element(&self, i: usize) -> Self::BaseField {
```

Comment:
I think the challenge with this approach is that we won't be able to use it generically when we want to get a base element out of a field element, but we don't know what the actual underlying field element is (e.g., see [here](https://github.com/facebook/winterfell/blob/77da6240fbd89854c2b4e25539657d5b7dea91ad/prover/src/matrix/col_matrix.rs#L113)).

Reply:
There is a way to do it, but it indeed starts to generate maybe unnecessary complication

```rust
trait FieldElement {
    type BaseElementSelector: From<usize>;
    ...
    fn base_element(&self, selector: Self::BaseElementSelector) -> Self::BaseField;
}
```

This `From<usize>` is quite hard to document as it might be a specific case requirement. It is a working alternative, but I'm not sure it's an actual improvement